### PR TITLE
Adds TLS check to mongodb destination and migrates util constants

### LINKED ISSUE #16279: Add ability to enforce SSL in MongoDB Destination connector
## Tell us about the problem you're trying to solve
MongoDB does not _always_ enforce SSL 

## Describe the solution you’d like
Implement configuration for adding `ssl` parameter for MongoDB destination

## Describe the alternative you’ve considered or used
A clear and concise description of any alternative solutions or features you've considered or are using today.

## Additional context

- [Connection Options](https://www.mongodb.com/docs/manual/reference/connection-string/#connection-options)
- [Java Driver documentation](http://mongodb.github.io/mongo-java-driver/3.0/driver/reference/connecting/ssl/)

## Are you willing to submit a PR?

Remove this with your answer :-)

### PR DESCRIPTION
**Pull request by ryankfu**
## What
Closes https://github.com/airbytehq/airbyte/issues/16279

## How
Follows https://github.com/airbytehq/airbyte/pull/17590/files which adds the ability to check for Mongodb `STANDALONE` instances of MongoDB and reaffirm that the connection when `tls` flag is present is marked as `true`

Migrates some of the shared constants into an already pre-existing MongoUtils file

## Recommended reading order
1. `MongodbDestinationStrictEncrypt.java`
2. `MongoUtils.java`
3. `MongodbDestinationStrictEncryptAcceptanceTest.java`
4. `MongodbDestination.java`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

Yes, customers who have the `tls` flag disabled or set to false will experience breaking changes when running their `check` method. This is expected and within reason as the user should not be able to set their connection to insecure traffic within Airbyte Cloud

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/destination-mongodb/src/main/java/io/airbyte/integrations/destination/mongodb/MongodbDestinationStrictEncrypt.java
Comment (ryankfu) [inline airbyte-integrations/connectors/destination-mongodb/src/main/java/io/airbyte/integrations/destination/mongodb/MongodbDestinationStrictEncrypt.java:21]: @tuliren just wanted to check, is there anything stopping me from moving the `MongodbDestinationStrictEncrypt` file from the `destination-mongodb-strict-encrypt` module instead of creating a copy of the `MongodbDestinationStrictEncrypt` in the non-strict-encrypt variant?

Currently hesitated to move the `strict-encrypt` variant since a duplicate file exists for the `source-postgres` and `source-postgres-strict-encrypt`

Comment (tuliren) [reply airbyte-integrations/connectors/destination-mongodb/src/main/java/io/airbyte/integrations/destination/mongodb/MongodbDestinationStrictEncrypt.java:21]: > moving the `MongodbDestinationStrictEncrypt` file from the `destination-mongodb-strict-encrypt` module

After moving the strict encrypt class from the `strict-encrypt` module to the normal module, will the `strict-encrypt` module still have a `mainClass` in its `build.gradle` file?  If yes, this would be the perfect solution for now. We won't have the duplicated class problem.

File: airbyte-integrations/connectors/destination-mongodb/src/main/java/io/airbyte/integrations/destination/mongodb/MongoDbDestinationUtils.java
Comment (akashkulk) [inline airbyte-integrations/connectors/destination-mongodb/src/main/java/io/airbyte/integrations/destination/mongodb/MongoDbDestinationUtils.java:10]: Is there a possibility of combining this with MongoDbSourceUtils to prevent code duplication?

File: airbyte-integrations/connectors/destination-mongodb-strict-encrypt/src/main/java/io.airbyte.integrations.destination.mongodb/MongodbDestinationStrictEncrypt.java
Comment (akashkulk) [inline airbyte-integrations/connectors/destination-mongodb-strict-encrypt/src/main/java/io.airbyte.integrations.destination.mongodb/MongodbDestinationStrictEncrypt.java:34]: If this isn't urgent - can you hold off until I merge the PR that adds the config exception handling? That way - you can just throw a ConfigErrorException instead

File: airbyte-integrations/connectors/destination-mongodb/src/main/java/io/airbyte/integrations/destination/mongodb/MongoDbDestinationUtils.java
Comment (ryankfu) [reply airbyte-integrations/connectors/destination-mongodb/src/main/java/io/airbyte/integrations/destination/mongodb/MongoDbDestinationUtils.java:10]: Good point, initially I didn't do that simply because it was in different packages but looking at this right now I see there's a `MongoUtils.java` where these changes and your changes can live. I'll work on moving the changes over to that utility file

File: airbyte-integrations/connectors/destination-mongodb-strict-encrypt/src/main/java/io.airbyte.integrations.destination.mongodb/MongodbDestinationStrictEncrypt.java
Comment (ryankfu) [reply airbyte-integrations/connectors/destination-mongodb-strict-encrypt/src/main/java/io.airbyte.integrations.destination.mongodb/MongodbDestinationStrictEncrypt.java:34]: Sure, I'll probably rework the `MongodbDestinationUtils` and `MongodbSourceUtils` into `MongoUtils.java` in the meantime. Your changes for that PR config exception should be ready to merge unless there's something else you needed to follow up on

Comment (akashkulk) [reply airbyte-integrations/connectors/destination-mongodb-strict-encrypt/src/main/java/io.airbyte.integrations.destination.mongodb/MongodbDestinationStrictEncrypt.java:34]: Yeah, there's still a degree of integration test mangling that I need to do for that PR. To unblock you and others I have https://github.com/airbytehq/airbyte/pull/18989 (which you already commented on)

I'll going to break that PR into smaller parts, but overall you should be unblocked on that front soon

Comment (akashkulk) [reply airbyte-integrations/connectors/destination-mongodb-strict-encrypt/src/main/java/io.airbyte.integrations.destination.mongodb/MongodbDestinationStrictEncrypt.java:35]: That PR is merged, so you should be able to throw a ConfigErrorException here now

**Review by akashkulk — COMMENTED** [review]
Thanks for the refactor! Some minor comments:
- You can throw a `ConfigErrorException` here now since that exception is handled in `IntegrationRunner` now
- Could you also just run the source-mongodb tests as well due to the refactor (apologies if you already ran them and I missed that github action)

**Review by akashkulk — APPROVED** [review]

**Review by akashkulk — APPROVED** [review]

**Comment by ryankfu** [comment]
/publish connector=connectors/destination-mongodb run-tests=false

**Comment by ryankfu** [comment]
/publish connector=connectors/destination-mongodb-strict-encrypt run-tests=false auto-bump-version=false

**Pull request merged**
